scripts/release.sh

```
@@ -116,6 +114,250 @@ if [ -z "$GITHUB_REMOTE" ]; then
     exit 1
 fi
 
+# Helper function to check if a tag exists as a release on GitHub
+check_github_release_exists() {
+    local tag_name=$1
+    local remote_url=$(git remote get-url "$GITHUB_REMOTE")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)